pkg/build/builder/common.go

```diff
@@ -201,15 +217,24 @@ func retryBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface,
 		latestBuild.Status.Output.To = build.Status.Output.To
 		latestBuild.Status.Stages = build.Status.Stages
 
-		if _, err := client.UpdateDetails(latestBuild); err != nil {
-			return err
+		_, err = client.UpdateDetails(latestBuild)
+		netErr, ok := err.(net.Error)
+
+		switch {
+		case err == nil:
+			return true, nil
+		case errors.IsConflict(err):
+			latestBuild = nil
+			return false, err
+		case ok && netErr.Temporary():
+			return false, err
```

Comment:
that's what "same for every else you are returning the error" means, yes :)

unless there's an error you do not want to retry, do not return the error.